### ISSUE TITLE
Ability to update filter without a fixed sample period

### ISSUE THREAD

**Issue by vincentl98** [jmagnuson/ahrs-rs#19]
Both Madgwick and Mahony filter implementations are using a fixed `sample_period`. However in non real-time systems, it is more accurate to directly pass the time delta between two measurements:

```rust
fn update(
        &mut self,
        gyroscope: &Vector3<N>,
        accelerometer: &Vector3<N>,
        magnetometer: &Vector3<N>,
    ) -> Result<&Quaternion<N>, &str> {
```

would become in that case:

```rust
fn update(
        &mut self,
        gyroscope: &Vector3<N>,
        accelerometer: &Vector3<N>,
        magnetometer: &Vector3<N>,
        dt : N, // dt: time delta between the previous measurements and this one
    ) -> Result<&Quaternion<N>, &str> {
```

**Comment by jmagnuson** [comment]
Agree-- I originally realized this only after publishing the API, and implemented the (admittedly hacky) [`field_access`](https://github.com/jmagnuson/ahrs-rs#field_access) feature flag to be able to mutate the period for each sample update without a breaking releases.

It would be nice to satisfy both fixed and dynamic sample periods, but without separate implementations I fear it might have a performance impact (e.g., passing in a `Option<N>` and doing the mutation check internally). So unless a more elegant solution can be determined, the best option is probably just to do like you suggest and force the caller to manage period.

Eventually, it would be nice to add implementations for Unscented, Extended, etc. Fortunately those should also be compatible with the updated `Ahrs` trait.

**Comment by msdrigg** [comment]
I am also interested in this feature. Could you have a second method `update_dynamic(self, gyroscope, accelerometer, magnetometer, dt)` that internally mutates the sample rate and then calls `update`. I know that this might not be the desired api, so I am happy to fork this and edit (just for my own personal use, I won't publish on crates.io).
